pkg/kubernetes/clientbuilder.go

```diff
@@ -52,12 +65,30 @@ func (cb *ClientBuilder) Build(validate bool) (kubernetes.Interface, error) {
 	}
 
 	if validate {
-		timeout := int64(5)
-		_, err := clientSet.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{
-			TimeoutSeconds: &timeout,
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		start := time.Now()
+
+		// wait for 0, 5, 15 (2*5+5), 35 (2*15+5) seconds total elapsed time
+		backoff := wait.Backoff{
+			Duration: 5 * time.Second,
+			Factor:   2,
+			Jitter:   0,
+			Steps:    4,
+			Cap:      35 * time.Second,
+		}
+		err = wait.ExponentialBackoffWithContext(ctx, backoff, func() (done bool, err error) {
+			_, err = clientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				cb.logger.Info(errors.Wrapf(err, "validation of connection to Kubernetes cluster %s failed (%.0f seconds elapsed since start), retrying", config.Host, time.Since(start).Seconds()))
```

Comment:
Isn't logger.Error is better here?

Reply:
If we do error logging here then our control-plane might be spammed with outputs on bad connectivity. I left this on error after consulting with @tobiscr but I can also change it. WDYT?